### DISCUSSION TITLE
How to determine when `cdk deploy` is run?

### DISCUSSION THREAD

**Discussion by rnag** [aws/aws-cdk#21486] [Q&A] [answered]
Hi there,

I have an interesting question, but I would like to know if there is any way within a CDK app to know which cdk subcommand was invoked. Specifically, I am just trying to distinguish between `cdk deploy` and between other similar commands, such as `cdk synth` and `cdk destroy`.

I think at a minimum, it would be good enough to know if we are only deploying a CDK app or not (i.e. if `cdk deploy` is being run).

### Use case/scenario

_I can briefly go into the specifics of why this is an important question for me:_
So in my use case, I basically create a custom CDK construct. Let's say this construct can sometimes take a long time to build/run - so for example it can take up to 5 minutes to resolve the construct, which is fine for the build/deploy step when `cdk deploy` is run. However, in other cases, such as when `cdk synth` is run, I would like to skip this sometimes expensive logic, and maybe just return with a dummy file or artifact. Hence, I was curious to know if there exists a way currently to determine if we are in a deploy-related scenario or not - such as for example when `cdk destroy` is run, we are only interested in cleaning up the stack and we won't need to run the whole construct logic in such a scenario.

**Comment by calavera** [comment]
I think what you need is to extract the building logic and put it in the `bundling` property in the asset. That way, the building logic will only be executed during build/deploy in the cdk. Look at the construct for Go Lambda functions, I think it's something similar:

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-lambda-go/lib/bundling.ts#L73-L83

**Comment by bestickley** [comment]
I'd hoped that `process.argv` would reveal this information but it only show information of the `"app"` command in cdk.json :(

**Reply by bestickley** [reply]
@calavera's answer is best. See more help here: https://dev.to/aws-builders/aws-cdk-fullstack-polyglot-with-asset-bundling-318h

**Comment by bestickley** [comment, marked as answer]
I'm not sure if this is new, but now you can use `Stack.of(this).bundlingRequired` boolean property to determine if you should bundle.
